File: app/src/main/java/geekbrians/slava_5655380/ui/mainactivity/AppFragmentPageAdapter.java
package geekbrians.slava_5655380.ui.mainactivity;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import geekbrians.slava_5655380.ui.fragments.moon.NotificationFragment;
import geekbrians.slava_5655380.ui.fragments.home.HomeFragment;

public class AppFragmentPageAdapter extends FragmentPagerAdapter {

  public AppFragmentPageAdapter(FragmentManager fm) {
    super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
  }

  @NonNull
  @Override
  public Fragment getItem(int position) {
    switch (position) {
      case 0:
        return new HomeFragment();
      case 1:
        return DashboardFragment.newInstance("Tony");
      case 2:
        return NotificationFragment.newInstance();
      case 3:
        return NotificationFragment.newInstance();
      case 4:
        return NotificationFragment.newInstance();
      default:
        throw new RuntimeException("Not supported");
    }
  }

  @Override
  public int getCount() {
    return 5;
  }
}

File: app/src/main/java/geekbrians/slava_5655380/ui/mainactivity/AppFragmentPageAdapter.kt
package geekbrians.slava_5655380.ui.mainactivity

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import geekbrians.slava_5655380.ui.fragments.DashboardFragment
import geekbrians.slava_5655380.ui.fragments.earth.EarthDashboardChildFragmentA
import geekbrians.slava_5655380.ui.fragments.earth.EarthDashboardChildFragmentB
import geekbrians.slava_5655380.ui.fragments.home.HomeFragment
import geekbrians.slava_5655380.ui.fragments.mars.MarsFragment
import geekbrians.slava_5655380.ui.fragments.moon.MoonDashboardChildFragmentA
import geekbrians.slava_5655380.ui.fragments.moon.MoonDashboardChildFragmentB
import geekbrians.slava_5655380.ui.fragments.space.SpaceFragment

class AppFragmentPageAdapter(fm: FragmentManager?) : FragmentPagerAdapter(
    fm!!, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT
) {

    val earthChildDashboardFragments = arrayListOf<Fragment>(
        EarthDashboardChildFragmentA.newInstance("Earth"),
        EarthDashboardChildFragmentB.newInstance("Earth"),
    )

    val moonChildDashboardFragments = arrayListOf<Fragment>(
        MoonDashboardChildFragmentA.newInstance("Moon"),
        MoonDashboardChildFragmentB.newInstance("Moon"),
    )

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> HomeFragment()
            1 -> DashboardFragment.newInstance(
                "Earth",
                earthChildDashboardFragments
            )
            2 -> DashboardFragment.newInstance(
                "Moon",
                moonChildDashboardFragments
            )
            3 -> MarsFragment.newInstance()
            4 -> SpaceFragment.newInstance()
            else -> throw RuntimeException("Not supported")
        }
    }

    override fun getCount(): Int {
        return 5
    }
}